src/commands/migration_tasks.py

```diff
@@ -0,0 +1,1358 @@
+        """
+        new_port_trans = portfolio_transaction if portfolio_transaction else destination_models.PortfolioTransaction()
+        new_port_trans.transaction_number = transaction_number
+        new_port_trans.transaction_date = transaction_date
+        new_port_trans.amount = amount
+        new_port_trans.purchase_c_value = purchase_c_value
+        new_port_trans.portfolio_id = portfolio_id
+        new_port_trans.security_id = security_id
+        new_port_trans.updated = updated
+        backend_session.add(new_port_trans)
+        return new_port_trans
+
+
+class MigrateFundsTask(AbstractMigrationTask):
+    """
+    Migration task for funds
+    """
+
+    def __init__(self):
+        """
+        Constructor
+        """
+        self.kiid_engine = self.get_kiid_engine()
+
+    def get_name(self):
+        return "funds"
+
+    def up_to_date(self, backend_session: Session) -> bool:
+        with Session(self.kiid_engine) as kiid_session:
+            backend_fund_count = backend_session.execute(statement="SELECT COUNT(id) FROM fund").fetchone()
+            kiid_fund_count = kiid_session.execute(statement="SELECT COUNT(ID) FROM FUND").fetchone()
+            return backend_fund_count >= kiid_fund_count
+
+    def migrate(self, backend_session: Session, timeout: datetime) -> int:
+        synchronized_count = 0
+        with Session(self.kiid_engine) as kiid_session:
+
+            fund_rows = self.list_fund_rows(kiid_session=kiid_session)
+            for fund_row in fund_rows:
+                fund_id = fund_row.ID
+
+                fund: destination_models.Fund = backend_session.query(destination_models.Fund) \
+                    .filter(destination_models.Fund.original_id == fund_id) \
+                    .one_or_none()
+
+                if not fund:
+                    fund = destination_models.Fund()
+                    fund.original_id = fund_id
+
+                if fund_row.URL_EN:
+                    fund.kiid_url_en = fund_row.URL_EN
+
+                if fund_row.URL_FI:
+                    fund.kiid_url_fi = fund_row.URL_FI
+
+                if fund_row.URL_SV:
+                    fund.kiid_url_sv = fund_row.URL_SV
+
+                if fund_row.RISK_LEVEL:
+                    fund.risk_level = fund_row.RISK_LEVEL
+
+                backend_session.add(fund)
+                synchronized_count = synchronized_count + 1
+
+            return synchronized_count
+
+    @staticmethod
+    def list_fund_rows(kiid_session: Session):
+        """
+        Lists rows from the kiid funds table
+        Args:
+            kiid_session: KIID database session
+
+        Returns: rows from the kiid funds table
+        """
+        risk_query = "SELECT TOP 1 risk_level FROM TextContent WHERE fund_name = NAME_KIID ORDER BY modification_time"
+        statement = f"SELECT ID, URL_FI, URL_SV, URL_EN, ({risk_query}) as RISK_LEVEL FROM FUND"
+        return kiid_session.execute(statement=statement)
```

Comment:
i didnt understand this query, maybe we could have somewhere some information about this one as well?

Reply:
Explained in Slack :)